lnd.go

```diff
@@ -535,21 +531,30 @@ func Main() error {
 			err := fmt.Errorf("Unable to start autopilot agent: %v",
 				err)
 			ltndLog.Error(err)
-			return err
+			return nil, err
 		}
 	}
 
 	if cfg.Watchtower.Active {
 		if err := tower.Start(); err != nil {
 			err := fmt.Errorf("Unable to start watchtower: %v", err)
 			ltndLog.Error(err)
-			return err
+			return nil, err
 		}
 		defer tower.Stop()
 	}
+	return activeChainControl, nil
+}
 
+// Main is the true entry point for lnd. This function is required since defers
+// created in the top-level scope of a main method aren't executed if os.Exit()
+// is called.
+func Main() error {
+	if _, err := InitDaemon(); err != nil {
```

Comment:
not sure this will actually work in practice. all of the defers in `InitDaemon` will be executed upon it returning, meaning all databases will be closed and all servers will be shutdown. Those defers need to happen inside the scope of `Main`